openhands/runtime/client/runtime.py

```diff
@@ -225,34 +260,39 @@ def _init_container(
                 self.runtime_container_image,
                 command=(
                     f'/openhands/miniforge3/bin/mamba run --no-capture-output -n base '
-                    'PYTHONUNBUFFERED=1 poetry run '
-                    f'python -u -m openhands.runtime.client.client {self._port} '
-                    f'--working-dir {sandbox_workspace_dir} '
+                    f'poetry run '
+                    f'python -u -m openhands.runtime.client.client {self._container_port} '
+                    f'--working-dir "{sandbox_workspace_dir}" '
                     f'{plugin_arg}'
                     f'--username {"openhands" if self.config.run_as_openhands else "root"} '
                     f'--user-id {self.config.sandbox.user_id} '
                     f'{browsergym_arg}'
                 ),
                 network_mode=network_mode,
                 ports=port_mapping,
-                working_dir='/openhands/code/',
+                working_dir='/openhands/code/',  # do not change this!
                 name=self.container_name,
                 detach=True,
-                environment={'DEBUG': 'true'} if self.config.debug else None,
+                environment=environment,
                 volumes=volumes,
             )
             self.log_buffer = LogBuffer(container)
             logger.info(f'Container started. Server url: {self.api_url}')
             return container
         except Exception as e:
-            logger.error('Failed to start container')
+            logger.error(
+                f'VVVVV Instance {self.instance_id} FAILED to start container VVVVV'
+            )
             logger.exception(e)
-            self.close(close_client=False)
+            logger.error(
+                '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^'
+            )
+            self.close(close_client=False, rm_all_containers=False)
```

Comment:
This came basically from @mamoodi in other PR and it is indeed only used for tests here with False as we don't want to remove containers of concurrently running tests.

Reply:
As noted below, could we use it in tests, not in the app? This will apply to users machines too.

Reply:
There is no difference on users' machines. The default behavior is still the same.

Reply:
Just to be clear--we still run `rm_all_containers` when in normal app mode, right? It's just skipped in test mode?

Reply:
The way I'm reading it, this line should have `rm_all_containers=True`, no?

Reply:
Ahh I guess because we're in `_init_container`, we don't want to do the `rm`?

Reply:
Perhaps we get confused by the other file with the same topic. But this is not in tests, it's in client runtime, that's why it applies to users. Before, we were trying to clean our containers on error, now we don't.

We're okay because we're still cleaning them at normal shutdown?

Reply:
Please look at the method itself and it should become clear that the containers get closed as before:
https://github.com/All-Hands-AI/OpenHands/blob/ac024eded131389b6aafa51ccd583b099bc9c257/openhands/runtime/client/runtime.py#L344

Reply:
Ok, after Slack discussion, I got the raised issue understood - it's late and I'm old! 🤣 
The param in the last .close is reverted now. @enyst